avogadro/qtplugins/cartoons/cartoons.cpp

```diff
@@ -0,0 +1,373 @@
+  bezier->identifier().molecule = &molecule;
+  bezier->identifier().type = Rendering::BondType;
+  geometry->addDrawable(bezier);
+
+  for (const auto& bone : backbone) {
+    bezier->addPoint(bone.pos, bone.color1, radius, bone.group);
+  }
+}
+
+void renderTube(const AtomsPairList& backbone, const Molecule& molecule,
+                Rendering::GroupNode& node, float radius, size_t id)
+{
+  GeometryNode* geometry = new GeometryNode;
+  node.addChild(geometry);
+
+  BSplineGeometry* bezier = new BSplineGeometry;
+  bezier->identifier().molecule = &molecule;
+  bezier->identifier().type = Rendering::BondType;
+  geometry->addDrawable(bezier);
+
+  for (const auto& bone : backbone) {
+    bezier->addPoint(bone.pos, bone.color1, radius, bone.group);
+  }
+}
+
+void renderCartoon(const AtomsPairList& backbone, const Molecule& molecule,
+                   Rendering::GroupNode& node, float radius, size_t id)
+{
+  GeometryNode* geometry = new GeometryNode;
+  node.addChild(geometry);
+
+  Cartoon* cartoon = new Cartoon;
+  cartoon->identifier().molecule = &molecule;
+  cartoon->identifier().type = Rendering::BondType;
+  geometry->addDrawable(cartoon);
+  for (const auto& bone : backbone) {
+    cartoon->addPoint(bone.pos, bone.color1, radius, bone.group);
+  }
+}
+
+void Cartoons::process(const Molecule& molecule, Rendering::GroupNode& node)
+{
+  m_group = &node;
+  if (m_showBackbone || m_showTrace || m_showTube || m_showRibbon ||
+      m_showCartoon || m_showRope) {
+    map<size_t, AtomsPairList> backbones;
+    if (molecule.residues().size() > 0) {
+      backbones = getBackboneByResidues(molecule);
+    }
+    if (backbones.size() == 0) {
+      backbones = getBackboneManually(molecule);
+    }
+    size_t i = 0;
+    for (const auto& group : backbones) {
+      const auto& backbone = group.second;
+      if (m_showBackbone) {
+        renderBackbone(backbone, molecule, node, 0.1f);
+      }
+      if (m_showTrace) {
+        renderTube(backbone, molecule, node, -0.15f, i);
+      }
+      if (m_showTube) {
+        renderTube(backbone, molecule, node, 0.15f, i);
+      }
+      if (m_showRibbon) {
+        renderCartoon(backbone, molecule, node, -1.0f * Cartoon::ELIPSE_RATIO,
+                      i);
+      }
+      if (m_showCartoon) {
+        renderCartoon(backbone, molecule, node, 1.0f, i);
+      }
+      if (m_showRope) {
+        renderRope(backbone, molecule, node, 1.0f, i);
+      }
+      ++i;
+    }
+  }
+}
+
+void Cartoons::processEditable(const QtGui::RWMolecule& molecule,
+                               Rendering::GroupNode& node)
+{}
+
+bool Cartoons::isEnabled() const
+{
+  return m_enabled;
+}
+
+void Cartoons::setEnabled(bool enable)
+{
+  m_enabled = enable;
+}
+
+QWidget* Cartoons::setupWidget()
+{
+  if (!m_setupWidget) {
+    m_setupWidget = new QWidget(qobject_cast<QWidget*>(parent()));
+    QVBoxLayout* v = new QVBoxLayout;
+    const char* boxesText[] = { "Backbone", "Trace",   "Tube",
```

Comment:
This doesn't work well with translation because the script that grabs the messages won't see these.

Try something like:
QStringList labels;
labels << tr("Backbone") << tr("Trace") << tr("Tube") << tr("Ribbon") << tr("Cartoon"), tr("Rope");

If it gets more complicated, we can use QtDesigner to make a UI file, which writes a lot of this code for  us.